### PR TITLE
Docs site with mkdocs  

### PR DESCRIPTION
**Pull request by swiftdiaries**
1. Adds documentation site with mkdocs
2. Adds a GitHub Action to publish the site

### CODE REVIEW

**Comment by sub-mod** [comment]
Thanks for PR.
I am wondering where we are going to host the docs site ? @petermattson  any ideas ?

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+site_name: 'MLBox'
+site_description: 'Documentation site for MLBox'
+site_author: 'Adhita Selvaraj <adhita.selvaraj@gmail.com>'
```
Comment (sub-mod) [inline mkdocs.yml:3]: I suggest we use an email `mlperf@googlegroups.com`
already being used in other projects >https://github.com/mlperf/logging/blob/master/setup.py#L27

File: README.md [hunk 2]
```diff
@@ -1,27 +1,36 @@
 # MLBox
 
-This is the MLBox Prototype. This is  still under construction, some parts probably don't work yet, or may have unexpected/inconsistent behaviours.
```
Comment (sub-mod) [inline README.md:3]: IMHO this change can be removed helps in tracking README changes in future .
Readme changes are not required for mkdocs PR.

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# MLBox 
+
+This is the MLBox 📦 Prototype. This is  still under construction, some parts probably don't work yet, or may have unexpected/inconsistent behaviours.
```
Comment (sub-mod) [inline docs/index.md:3]: IMHO I think we should start the first docs with correct contents. 
index.md with incorrect changes can be avoided for mkdocs PR.

File: mkdocs.yml (hunk 1 above)
Comment (swiftdiaries) [reply mkdocs.yml]: Changed

File: README.md (hunk 2 above)
Comment (swiftdiaries) [reply README.md]: Addressed

File: docs/index.md (hunk 3 above)
Comment (swiftdiaries) [reply docs/index.md]: Addressed

File: .github/workflows/docs-site.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    steps:
+      - uses: actions/checkout@master
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@master
+        with:
+          python-version: 3.6
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
```
Comment (sub-mod) [inline .github/workflows/docs-site.yml:27]: The location of mkdocs requirements.txt conflicts with projects  requirements.txt.
Is it a good idea to move it to docs folder ? What do you think is the best approach ?

Comment (swiftdiaries) [reply .github/workflows/docs-site.yml]: Added a separate `mkdocs-requirements.txt` for mkdocs related dependencies

**Comment by sub-mod** [comment]
@swiftdiaries can you rebase to a single commit?
I am unable to merge `This branch cannot be rebased due to conflicts`

**Comment by swiftdiaries** [comment]
Addressed merge conflicts and pulled changes from latest master

**Comment by swiftdiaries** [comment]
Closing this PR in favor of https://github.com/mlperf/mlbox/pull/60

**Pull request closed without merging**